### PR TITLE
[API] LR scheduler support

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

This PR adds LR scheduler support. It adds also a nice CI test 

cc @lvwerra

### CODE REVIEW

File: trl/trainer/ppo_trainer.py [hunk 1]
```diff
@@ -390,6 +406,7 @@ def step(
             stats = self.gather_stats(stats)
         stats = stats_to_np(stats)
         timing["time/ppo/calc_stats"] = time.time() - t
+        stats["learning_rate"] = self.optimizer.param_groups[0]["lr"]
```
Comment (lvwerra) [inline trl/trainer/ppo_trainer.py:409]: what do you think about adding a prefix which means it gets ordered in wandb logs?

```suggestion
        stats["ppo/learning_rate"] = self.optimizer.param_groups[0]["lr"]
```

**Review by lvwerra — COMMENTED** [review]
Looks great, thanks for adding this! Left a few small comments, otherwise good to merge :)

File: trl/trainer/ppo_trainer.py [hunk 2]
```diff
@@ -162,13 +166,25 @@ def __init__(
...
 
-        self.model, self.ref_model, self.optimizer, self.data_collator, self.dataloader = self.accelerator.prepare(
-            self.model, self.ref_model, self.optimizer, self.data_collator, self.dataloader
+        (
+            self.model,
+            self.ref_model,
+            self.optimizer,
+            self.data_collator,
+            self.dataloader,
+            self.lr_scheduler,
```
[the remark this answers is not in the document]
Comment (younesbelkada) [reply trl/trainer/ppo_trainer.py:185]: Yes it works!

File: tests/test_ppo_trainer.py [hunk 3]
```diff
@@ -162,6 +163,46 @@ def test_ppo_step_with_no_ref_sgd(self):
         for stat in EXPECTED_STATS:
             assert stat in train_stats.keys()
 
+    def test_ppo_step_with_no_ref_sgd_lr_scheduler(self):
```
[the remark this answers is not in the document]
Comment (younesbelkada) [reply tests/test_ppo_trainer.py:166]: yes I can add that ;) !

[the remark this answers is not in the document]
Comment (younesbelkada) [reply tests/test_ppo_trainer.py:166]: fixed in [f930f9b](https://github.com/lvwerra/trl/pull/96/commits/f930f9b58f20d552b62ef8c26d835fd643896fbb)

**Review by lvwerra — APPROVED** [review]
